src/main/java/com/elixr/poc/service/UserService.java

```diff
@@ -21,31 +20,33 @@ public class UserService {
 
     public UserService(UserRepository userRepository) {
         this.userRepository = userRepository;
-
     }
 
     /**
      * Deleting the user by the userId.
-     * throwing a NoRecordFoundException to handel if the UserId is not present.
+     * Throwing a NoRecordFoundException to handel if the UserId is not present.
+     * @param userId
+     * @return
+     * @throws GlobalException
      */
-    public boolean deleteUserDetails(UUID userId) throws NoRecordFoundException {
+    public boolean deleteUserDetails(UUID userId) throws GlobalException {
         boolean success = false;
         boolean userRecordExists = userRepository.existsById(userId);
         if (userRecordExists) {
             userRepository.deleteById(userId);
             success = true;
         } else {
-            throw new NoRecordFoundException(ApplicationConstants.ID_MISMATCH);
+            throw new GlobalException(ApplicationConstants.ID_MISMATCH);
```

Comment:
Error message not matching with the specific use case. Here the scenario is the id does not exist, not the id mismatch. Also, the GlobalException class is a custom exception class and it is better to rename it to be specific to your application

Reply:
Updated as suggested